cf-oscore/src/test/java/org/eclipse/californium/oscore/OSCoreOuterBlockwiseTest.java

```diff
@@ -0,0 +1,567 @@
+		} catch (OSException e) {
+			System.err.println("Failed to set server OSCORE Context information!");
+		}
+	}
+
+	/**
+	 * Create a simple OSCORE server that supports block-wise.
+	 * 
+	 * @param serverResponseBlockwise the server responds with block-wise
+	 */
+	private void createOscoreServer(boolean serverResponseBlockwise) {
+
+		setServerContext();
+
+		CoapEndpoint.Builder builder = new CoapEndpoint.Builder();
+		if (serverResponseBlockwise) {
+			builder.setNetworkConfig(blockwiseConfig);
+		}
+		builder.setInetSocketAddress(TestTools.LOCALHOST_EPHEMERAL);
+		builder.setCoapStackFactory(new OSCoreCoapStackFactory());
+		builder.setCustomCoapStackArgument(dbServer);
+		CoapEndpoint serverEndpoint = builder.build();
+
+		CoapServer server = new CoapServer();
+		cleanup.add(server);
+		server.addEndpoint(serverEndpoint);
+		resource = new MyResource(TARGET);
+		server.add(resource);
+		server.start();
+
+		serverUri = TestTools.getUri(serverEndpoint, TARGET);
+	}
+
+	private static String createRandomPayload(int size) {
+		StringBuilder builder = new StringBuilder(size);
+		Random random = new Random(size);
+		for (int i = 0; i < size; ++i) {
+			builder.append(random.nextInt(10));
+		}
+		return builder.toString();
+	}
+
+	private static class MyResource extends CoapResource {
+
+		/**
+		 * Request counter. Ensure, that transparent blockwise is not accidently
+		 * split into "intermediary block" requests.
+		 */
+		private final AtomicInteger counter = new AtomicInteger();
+		private volatile String currentPayload;
+
+		public MyResource(String name) {
+			super(name);
+		}
+
+		@Override
+		public void handleGET(CoapExchange exchange) {
+			counter.incrementAndGet();
+			Response response = new Response(ResponseCode.CONTENT);
+			response.setPayload(currentPayload);
+			exchange.respond(response);
+		}
+
+		@Override
+		public void handlePUT(CoapExchange exchange) {
+			counter.incrementAndGet();
+			currentPayload = exchange.getRequestText();
+			Response response = new Response(ResponseCode.CHANGED);
+			exchange.respond(response);
+		}
+
+		@Override
+		public void handlePOST(CoapExchange exchange) {
+			counter.incrementAndGet();
+			currentPayload += exchange.getRequestText();
+			Response response = new Response(ResponseCode.CONTENT);
+			response.setPayload(currentPayload);
+			exchange.respond(response);
+		}
+
+		public void setPayload(String payload) {
+			currentPayload = payload;
+		}
+
+		public int getCounter() {
+			return counter.get();
+		}
+	}
+
+	/**
+	 * Create simple non-OSCORE proxy.
+	 * 
+	 * @param proxyRequestBlockwise the proxy sends requests with block-wise
+	 * @param proxyResponseBlockwiseEnabled the proxy responds with block-wise
+	 */
+	private void createSimpleProxy(final boolean proxyRequestBlockwise, final boolean proxyResponseBlockwiseEnabled) {
+
+		final Coap2CoapTranslator coapTranslator = new Coap2CoapTranslator();
+		final CoapUriTranslator uriTranslator = new CoapUriTranslator();
```

Comment:
A `Coap2CoapTranslator` extends a `CoapUriTranslator`. Therefore you may replace the separate `uriTranslator` and just use the `coapTranslator` instead.

Reply:
I see, I did not notice that. I have now updated the code accordingly in commit 4886d668163d16f0597577e74fe94c7c8a1535fe